# Unmapped MARC fields and subfields
This notebook:
* Documents MARC fields and subfields omitted from Argot mappings
* Identifies MARC mapping decisions to be made, by analyzing:
 * current MARC specification from http://www.loc.gov/marc/bibliographic/ecbdlist.html 
 * Argot mappings from https://github.com/trln/data-documentation/blob/master/argot/argot.xlsx
 * Our explicit field and subfield omission decisions 

MARC fields we have decided to exclude from Argot mappings are recorded in [`unmapped_marc_tags.json`](https://github.com/trln/data-documentation/blob/master/marc/unmapped_marc_tags.json).

Specific MARC subfields excluded from Argot mappings *when some subfields from the field ARE mapped* are recorded in [`unmapped_marc_subfields.json`](https://github.com/trln/data-documentation/blob/master/marc/unmapped_marc_subfields.json).

## Use the current MARC specification
First, run script to generate JSON hash of current MARC specification. If successful: 
* Out will = true
* `marc_definition.json` is written to `data-documentation/marc` folder

In [63]:
require './extract_marc_bib_spec'

false

## Get all our mappings data
This next bit just reads in all the data we'll need: 
* `marc_spec` is the updated MARC specification created in the previous step
* `unmapped_fields` is our explicit decisions about MARC fields not mapped to Argot
* `unmapped_subfields` is our explicit decisions about individual MARC subfields not mapped to Argot. These subfields all occur in MARC fields that are otherwise mapped to Argot. 

In [64]:
marc_spec = JSON.parse(File.read('marc_definition.json'))
unmapped_fields = JSON.parse(File.read('unmapped_marc_tags.json'))
unmapped_subfields = JSON.parse(File.read('unmapped_marc_subfields.json'))

{"567"=>{"b"=>{"omission_type"=>"temporary", "omission_reason_category"=>"low priority", "omission_details"=>"UNC has no data in this subfield. Consider mapping to genre_headings and subject_genre if ever meaningfully populated"}}, "022"=>{"m"=>{"omission_type"=>nil, "omission_reason_category"=>nil, "omission_details"=>nil}, "y"=>{"omission_type"=>nil, "omission_reason_category"=>nil, "omission_details"=>nil}, "z"=>{"omission_type"=>nil, "omission_reason_category"=>nil, "omission_details"=>nil}}}

Next get our current MARC-to-Argot mappings from Argot spreadsheet and turn this into a hash we can use:

In [65]:
require 'simple_xlsx_reader'
doc = SimpleXlsxReader.open('../argot/argot.xlsx')
mappings_d = doc.sheets[1].data
mappings_h = doc.sheets[1].headers
marc_only = mappings_d.select{ |r| r[2] == "MARC"}.map{ |r| [r[5].rjust(3,'0'), r[6], r[3]] }
# Remove mappings for fixed fields and those where data is a constant
marc_only.reject!{ |r| r[0] =~ /(00[678]|\{na\})/}
# Change {na}, {varies}, period, space, or parentheses in subfield list to nothing
marc_only.map!{ |r| [r[0], r[1].gsub(/\{(na|varies)\}|[.() ]/, ''), r[2]]}
# Convert subfield list to array
marc_only.map!{ |r| [r[0], r[1].split(''), r[2]]}
# explode mappings list, so there's one array per subfield value
# orig: ["111", ["j", "4"], "n"]
# new: ["111", "j", "n"], ["111", 4", "n"]
flat_mappings = []
marc_only.each do |r|  
  r[1].each { |sf| flat_mappings << [r[0], sf, r[2]] }
end
mapping_hash = {}
flat_mappings.uniq!.each do |r|
  if mapping_hash.has_key?(r[0])
    mapping_hash[r[0]][r[1]] = r[2]
  else
    mapping_hash[r[0]] = {r[1] => r[2]}
  end
end
puts JSON.pretty_generate(mapping_hash)

{
  "581": {
    "a": "n",
    "z": "n",
    "3": "n"
  },
  "790": {
    "a": "n",
    "b": "n",
    "c": "n",
    "d": "n",
    "g": "n",
    "q": "n",
    "u": "n"
  },
  "791": {
    "a": "n",
    "b": "n",
    "c": "n",
    "d": "n",
    "f": "n",
    "g": "n"
  },
  "250": {
    "3": "n",
    "a": "n",
    "b": "n"
  },
  "254": {
    "a": "n"
  },
  "310": {
    "a": "n",
    "b": "n"
  },
  "321": {
    "a": "n",
    "b": "n"
  },
  "382": {
    "a": "y",
    "b": "y",
    "d": "y",
    "p": "y"
  },
  "384": {
    "a": "y"
  },
  "567": {
    "b": "n",
    "a": "n"
  },
  "653": {
    "a": "n"
  },
  "655": {
    "a": "y",
    "v": "n",
    "x": "y",
    "y": "n",
    "z": "n"
  },
  "599": {
    "b": "y"
  },
  "260": {
    "3": "n",
    "a": "n",
    "b": "n",
    "c": "n",
    "e": "n",
    "f": "n",
    "g": "n"
  },
  "264": {
    "3": "n",
    "a": "n",
    "b": "n",
    "c": "n"
  },
  "700": {
    "a": "n",
    "b": "n",
    "c": "n",
    "d": "n",
    "g": "n",
    "j

```
"581": {
    "a": "n",
    "z": "n",
    "3": "n"
  }
```

Subfields a, z, and 3 from the 581 field are mapped to Argot. 

The "n" at the end of each subfield line means these are final mappings for production. "y" in this position indicates a provisional mapping, which may not yet be implemented or which is subject to major changes.

## Unmapped MARC, field tag level

The block below reports on the fields we've explicitly decided not to map.

In [66]:
unmapped_tag_table = { 'MARC tag' => [],
                       'Name' => [],
                       'Omission type' => [],
                       'Omission category' => [],
                       'Details' => []
                     }
unmapped_fields.sort.each do |tag, info|
  unmapped_tag_table['MARC tag'] << tag
  unmapped_tag_table['Name'] << marc_spec[tag]['name']
  unmapped_tag_table['Omission type'] << info['omission_type']
  unmapped_tag_table['Omission category'] << info['omission_reason_category']
  unmapped_tag_table['Details'] << info['omission_details']
end
IRuby.display IRuby.table(unmapped_tag_table, maxrows: 100)

"<table><tr><th>MARC tag</th><th>Name</th><th>Omission type</th><th>Omission category</th><th>Details</th></tr><tr><td>306</td><td>PLAYING TIME </td><td>permanent</td><td>data unusable</td><td>Usually duplicates data recorded in 300 and/or 505. Recorded in a cryptic manner and there's no way to specify which duration recorded goes with which part of the described item, if it's a multi-part thing</td></tr><tr><td>307</td><td>HOURS, ETC. </td><td>permanent</td><td>irrelevant</td><td>No fields in UNC catalog. Hours an item is available is kind of a weird concept in the catalog, too.</td></tr><tr><td>336</td><td>CONTENT TYPE </td><td>temporary</td><td>low priority</td><td>Terms used are not user-friendly; this information is usually better expressed elsewhere in the record</td></tr><tr><td>337</td><td>MEDIA TYPE </td><td>temporary</td><td>low priority</td><td>Terms used are not user-friendly; this information is usually better expressed elsewhere in the record</td></tr><tr><td>338</td><td>CARRIER TYPE </td><td>temporary</td><td>low priority</td><td>Terms used are not user-friendly; this information is usually better expressed elsewhere in the record</td></tr><tr><td>342</td><td>GEOSPATIAL REFERENCE DATA </td><td>temporary</td><td>low priority</td><td>Complex data mapping; appears in <2000 records in UNC catalog on 2018-05-21. Need to assess if this adds useful info, and if so, how to map.</td></tr><tr><td>343</td><td>PLANAR COORDINATE DATA </td><td>temporary</td><td>infrequently used</td><td>only 2 records in UNC catalog</td></tr><tr><td>348</td><td>FORMAT OF NOTATED MUSIC </td><td>temporary</td><td>low priority</td><td>Look into whether it should map to genre</td></tr><tr><td>355</td><td>SECURITY CLASSIFICATION CONTROL </td><td>permanent</td><td>irrelevant</td><td>No records in UNC catalog</td></tr><tr><td>357</td><td>ORIGINATOR DISSEMINATION CONTROL </td><td>permanent</td><td>irrelevant</td><td>No records in UNC catalog</td></tr><tr><td>363</td><td>NORMALIZED DATE AND SEQUENTIAL DESIGNATION </td><td>permanent</td><td>data unusable</td><td>Holdings-like hell format specifying exact dates; only 10 in UNC catalog, most look like they should be 362s</td></tr><tr><td>370</td><td>ASSOCIATED PLACE </td><td>temporary</td><td>infrequently used</td><td>22 fields in UNC catalog -- maybe one day map to a subject-ish geographical entry or something?</td></tr><tr><td>377</td><td>ASSOCIATED LANGUAGE </td><td>temporary</td><td>infrequently used</td><td>1 field in UNC catalog -- maybe one day map to language facet if it adds unique info</td></tr><tr><td>380</td><td>FORM OF WORK </td><td>temporary</td><td>low priority</td><td>Look into mapping to genre</td></tr><tr><td>381</td><td>OTHER DISTINGUISHING CHARACTERISTICS OF WORK OR EXPRESSION </td><td>permanent</td><td>data unusable</td><td>Mishmash of different kinds of data issuing body, arranged statement of music, version, or a geographic term with no coding that tells you what the data is</td></tr><tr><td>383</td><td>NUMERIC DESIGNATION OF MUSICAL WORK </td><td>permanent</td><td>duplicate data</td><td>Currently duplicates data recorded in other parts of the record, particularly 130 and 240 fields.</td></tr><tr><td>385</td><td>AUDIENCE CHARACTERISTICS </td><td>temporary</td><td>low priority</td><td>Look into mapping to `note_general` with label `For audiences`</td></tr><tr><td>386</td><td>CREATOR/CONTRIBUTOR CHARACTERISTICS </td><td>temporary</td><td>infrequently used</td><td>Could eventually leverage in some \"Works by\" field, but only 248 fields in UNC's catalog today 2018-05-18</td></tr><tr><td>388</td><td>TIME PERIOD OF CREATION </td><td>temporary</td><td>infrequently used</td><td>Only 3 fields in UNC catalog on 2018-05-18. Eventually could be mapped as a chronological subect or something. </td></tr></table>"

#<CZTop::Socket::PUB:0x1ac82b0 last_endpoint="tcp://127.0.0.1:59847">

And the following lists all other **non-obsolete, non-local** unmapped MARC variable fields we haven't made an explicit decision about:

In [67]:
unmapped = {}
marc_spec.each do |tag, info|
  if mapping_hash.has_key?(tag)
    next
  elsif unmapped_fields.has_key?(tag)
    next
  elsif tag =~ /00[356789]/
    next
  elsif info['context'] =~ /OBSOLETE|LOCAL/
  else
    unmapped[tag] = {'name' => info['name'].strip,
                     'context' => info['context']
      }
  end
end

table2 = { 'MARC tag' => [],
           'Description' => [],
           'Note' => []
         }
unmapped.sort.each do |tag, info|
  table2['MARC tag'] << tag
  table2['Description'] << info['name']
  table2['Note'] << info['context']
end
IRuby.display IRuby.table(table2, maxrows: 100)

"<table><tr><th>MARC tag</th><th>Description</th><th>Note</th></tr><tr><td>001</td><td>CONTROL NUMBER</td><td></td></tr><tr><td>013</td><td>PATENT CONTROL INFORMATION</td><td></td></tr><tr><td>016</td><td>NATIONAL BIBLIOGRAPHIC AGENCY CONTROL NUMBER</td><td></td></tr><tr><td>017</td><td>COPYRIGHT OR LEGAL DEPOSIT NUMBER</td><td></td></tr><tr><td>018</td><td>COPYRIGHT ARTICLE-FEE CODE</td><td></td></tr><tr><td>025</td><td>OVERSEAS ACQUISITION NUMBER</td><td></td></tr><tr><td>026</td><td>FINGERPRINT IDENTIFIER</td><td></td></tr><tr><td>031</td><td>MUSICAL INCIPITS INFORMATION</td><td></td></tr><tr><td>032</td><td>POSTAL REGISTRATION NUMBER</td><td></td></tr><tr><td>033</td><td>DATE/TIME AND PLACE OF AN EVENT</td><td></td></tr><tr><td>034</td><td>CODED CARTOGRAPHIC MATHEMATICAL DATA</td><td></td></tr><tr><td>036</td><td>ORIGINAL STUDY NUMBER FOR COMPUTER DATA FILES</td><td></td></tr><tr><td>037</td><td>SOURCE OF ACQUISITION</td><td></td></tr><tr><td>038</td><td>RECORD CONTENT LICENSOR</td><td></td></tr><tr><td>040</td><td>CATALOGING SOURCE</td><td></td></tr><tr><td>042</td><td>AUTHENTICATION CODE</td><td></td></tr><tr><td>043</td><td>GEOGRAPHIC AREA CODE</td><td></td></tr><tr><td>044</td><td>COUNTRY OF PUBLISHING/PRODUCING ENTITY CODE</td><td></td></tr><tr><td>045</td><td>TIME PERIOD OF CONTENT</td><td></td></tr><tr><td>046</td><td>SPECIAL CODED DATES</td><td></td></tr><tr><td>047</td><td>FORM OF MUSICAL COMPOSITION CODE</td><td></td></tr><tr><td>048</td><td>NUMBER OF MUSICAL INSTRUMENTS OR VOICES CODE</td><td></td></tr><tr><td>050</td><td>LIBRARY OF CONGRESS CALL NUMBER</td><td></td></tr><tr><td>051</td><td>LIBRARY OF CONGRESS COPY, ISSUE, OFFPRINT STATEMENT</td><td></td></tr><tr><td>052</td><td>GEOGRAPHIC CLASSIFICATION</td><td></td></tr><tr><td>055</td><td>CLASSIFICATION NUMBERS ASSIGNED IN CANADA</td><td></td></tr><tr><td>060</td><td>NATIONAL LIBRARY OF MEDICINE CALL NUMBER</td><td></td></tr><tr><td>061</td><td>NATIONAL LIBRARY OF MEDICINE COPY STATEMENT</td><td></td></tr><tr><td>066</td><td>CHARACTER SETS PRESENT</td><td></td></tr><tr><td>070</td><td>NATIONAL AGRICULTURAL LIBRARY CALL NUMBER</td><td></td></tr><tr><td>071</td><td>NATIONAL AGRICULTURAL LIBRARY COPY STATEMENT</td><td></td></tr><tr><td>072</td><td>SUBJECT CATEGORY CODE</td><td></td></tr><tr><td>080</td><td>UNIVERSAL DECIMAL CLASSIFICATION NUMBER</td><td></td></tr><tr><td>082</td><td>DEWEY DECIMAL CLASSIFICATION NUMBER</td><td></td></tr><tr><td>083</td><td>ADDITIONAL DEWEY DECIMAL CLASSIFICATION NUMBER</td><td></td></tr><tr><td>084</td><td>OTHER CLASSIFICATION NUMBER</td><td></td></tr><tr><td>085</td><td>SYNTHESIZED CLASSIFICATION NUMBER COMPONENTS</td><td></td></tr><tr><td>086</td><td>GOVERNMENT DOCUMENT CLASSIFICATION NUMBER</td><td></td></tr><tr><td>242</td><td>TRANSLATION OF TITLE BY CATALOGING AGENCY</td><td></td></tr><tr><td>243</td><td>COLLECTIVE UNIFORM TITLE</td><td></td></tr><tr><td>256</td><td>COMPUTER FILE CHARACTERISTICS</td><td></td></tr><tr><td>257</td><td>COUNTRY OF PRODUCING ENTITY</td><td></td></tr><tr><td>258</td><td>PHILATELIC ISSUE DATE</td><td></td></tr><tr><td>263</td><td>PROJECTED PUBLICATION DATE</td><td></td></tr><tr><td>270</td><td>ADDRESS</td><td></td></tr><tr><td>365</td><td>TRADE PRICE</td><td></td></tr><tr><td>366</td><td>TRADE AVAILABILITY INFORMATION</td><td></td></tr><tr><td>524</td><td>PREFERRED CITATION OF DESCRIBED MATERIALS NOTE</td><td></td></tr><tr><td>530</td><td>ADDITIONAL PHYSICAL FORM AVAILABLE NOTE</td><td></td></tr><tr><td>540</td><td>TERMS GOVERNING USE AND REPRODUCTION NOTE</td><td></td></tr><tr><td>542</td><td>INFORMATION RELATING TO COPYRIGHT STATUS</td><td></td></tr><tr><td>552</td><td>ENTITY AND ATTRIBUTE INFORMATION NOTE</td><td></td></tr><tr><td>565</td><td>CASE FILE CHARACTERISTICS NOTE</td><td></td></tr><tr><td>583</td><td>ACTION NOTE</td><td></td></tr><tr><td>584</td><td>ACCUMULATION AND FREQUENCY OF USE NOTE</td><td></td></tr><tr><td>588</td><td>SOURCE OF DESCRIPTION NOTE</td><td></td></tr><tr><td>654</td>

#<CZTop::Socket::PUB:0x1ac82b0 last_endpoint="tcp://127.0.0.1:59847">

## Unmapped MARC, subfield level
Here, we are looking at subfields in MARC fields that are at least partly mapped to Argot. 
What subfields in such partially mapped fields are **not** mapped to Argot? 

### Explicit subfield exclusion decisions:

In [68]:
unmapped_sf_table = { 'MARC tag' => [],
                      'Subfield code' => [],
                       'Field name' => [],
                       'Subfield name' => [],
                       'Omission type' => [],
                       'Omission category' => [],
                       'Details' => []
                     }
unmapped_subfields.sort.each do |tag, sfs|
  sfs.each do |code, info|
    unmapped_sf_table['MARC tag'] << tag
    unmapped_sf_table['Subfield code'] << code
    unmapped_sf_table['Field name'] << marc_spec[tag]['name']
    unmapped_sf_table['Subfield name'] << marc_spec[tag]['subfields'][code]['name']
    unmapped_sf_table['Omission type'] << info['omission_type']
    unmapped_sf_table['Omission category'] << info['omission_reason_category']
    unmapped_sf_table['Details'] << info['omission_details']
  end
end
IRuby.display IRuby.table(unmapped_sf_table, maxrows: 100)

"<table><tr><th>MARC tag</th><th>Subfield code</th><th>Field name</th><th>Subfield name</th><th>Omission type</th><th>Omission category</th><th>Details</th></tr><tr><td>022</td><td>m</td><td>INTERNATIONAL STANDARD SERIAL NUMBER </td><td>Canceled ISSN-L</td><td></td><td></td><td></td></tr><tr><td>022</td><td>y</td><td>INTERNATIONAL STANDARD SERIAL NUMBER </td><td>Incorrect ISSN</td><td></td><td></td><td></td></tr><tr><td>022</td><td>z</td><td>INTERNATIONAL STANDARD SERIAL NUMBER </td><td>Canceled ISSN</td><td></td><td></td><td></td></tr><tr><td>567</td><td>b</td><td>METHODOLOGY NOTE </td><td>Controlled term</td><td>temporary</td><td>low priority</td><td>UNC has no data in this subfield. Consider mapping to genre_headings and subject_genre if ever meaningfully populated</td></tr></table>"

#<CZTop::Socket::PUB:0x1ac82b0 last_endpoint="tcp://127.0.0.1:59847">

### Unmapped subfields from MARC specification
These subfields may have been overlooked. If an explicit decision has been made to exclude this subfield from mapping to Argot, that decision should be documented in https://github.com/trln/data-documentation/blob/master/marc/unmapped_marc_subfields.json

Ignores: 
* Obsolete or local subfields
* Subfield 0 where subfield name = Authority record control number or standard number
* Subfield 1 where subfield name = Real World Object URI
* Subfield 2 where subfield name starts with Source
* Subfield 6 where subfield name = Linkage
* Subfield 8 where subfield name = Field link and sequence number

First we flatten out all subfield data so it's easier to work with:

In [86]:
subfields = []
marc_spec.each do |tag, info|
  if info['subfields'].size > 0
    info['subfields'].each do |code, sfinfo|
      subfields << {'tag' => tag,
        'code' => code,
        'name' => "#{info['name']} - #{sfinfo['name']}",
        'context' => sfinfo['context']}      
    end
  end
end

puts subfields.size

2491


Then, get rid of subfields that are in unmapped MARC fields. Keep only those whose field tag appears in our mappings spreadsheet.

In [87]:
subfields.select!{ |sf| mapping_hash.has_key?(sf['tag']) }
puts subfields.size

1491


Then, remove:
* Obsolete or local subfields
* Subfield 0 where subfield name = Authority record control number or standard number
* Subfield 1 where subfield name = Real World Object URI
* Subfield 2 where subfield name starts with Source
* Subfield 6 where subfield name = Linkage
* Subfield 7 where subfield name = Control subfield
* Subfield 8 where subfield name = Field link and sequence number

In [88]:
subfields.reject!{ |sf| sf['code'] == '0' && sf['name'] =~ / - Authority record control number/  }
subfields.reject!{ |sf| sf['code'] == '1' && sf['name'] =~ / - Real World Object URI/ }
subfields.reject!{ |sf| sf['code'] == '2' && sf['name'] =~ / - Source/ }
subfields.reject!{ |sf| sf['code'] == '6' && sf['name'] =~ / - Linkage/ }
subfields.reject!{ |sf| sf['code'] == '7' && sf['name'] =~ / - Control subfield/ }
subfields.reject!{ |sf| sf['code'] == '8' && sf['name'] =~ / - Field link and sequence number/ }
subfields.reject!{ |sf| sf['context'] =~ /OBSOLETE|LOCAL/ }
puts subfields.size

1102


Then, remove subfields we have mapped in Argot spreadsheet:

In [89]:
subfields.reject!{ |sf| mapping_hash.has_key?(sf['tag']) && mapping_hash[sf['tag']].has_key?(sf['code']) }
puts subfields.size

150


Then, output the remaining unmapped subfields to a table for analysis:

In [90]:
table4 = { 'MARC tag' => [],
  'subfield code' => [],
  'name' => []
  }

subfields.each do |sf|
  table4['MARC tag'] << sf['tag']
  table4['subfield code'] << sf['code']
  table4['name'] << sf['name']
end
IRuby.display IRuby.table(table4, maxrows: 200)

"<table><tr><th>MARC tag</th><th>subfield code</th><th>name</th></tr><tr><td>015</td><td>z</td><td>NATIONAL BIBLIOGRAPHY NUMBER  - Canceled/Invalid national bibliography number</td></tr><tr><td>020</td><td>c</td><td>INTERNATIONAL STANDARD BOOK NUMBER  - Terms of availability</td></tr><tr><td>022</td><td>m</td><td>INTERNATIONAL STANDARD SERIAL NUMBER  - Canceled ISSN-L</td></tr><tr><td>022</td><td>y</td><td>INTERNATIONAL STANDARD SERIAL NUMBER  - Incorrect ISSN</td></tr><tr><td>022</td><td>z</td><td>INTERNATIONAL STANDARD SERIAL NUMBER  - Canceled ISSN</td></tr><tr><td>024</td><td>c</td><td>OTHER STANDARD IDENTIFIER  - Terms of availability</td></tr><tr><td>028</td><td>a</td><td>PUBLISHER NUMBER OR DISTRIBUTOR NUMBER  - Publisher or distributor number</td></tr><tr><td>035</td><td>z</td><td>SYSTEM CONTROL NUMBER  - Canceled/invalid control number</td></tr><tr><td>041</td><td>b</td><td>LANGUAGE CODE  - Language code of summary or abstract</td></tr><tr><td>041</td><td>f</td><td>LANGUAGE CODE  - Language code of table of contents</td></tr><tr><td>041</td><td>h</td><td>LANGUAGE CODE  - Language code of original</td></tr><tr><td>041</td><td>j</td><td>LANGUAGE CODE  - Language code of subtitles or captions</td></tr><tr><td>041</td><td>k</td><td>LANGUAGE CODE  - Language code of intermediate translations</td></tr><tr><td>041</td><td>m</td><td>LANGUAGE CODE  - Language code of original accompanying materials other than librettos</td></tr><tr><td>041</td><td>n</td><td>LANGUAGE CODE  - Language code of original libretto</td></tr><tr><td>210</td><td>b</td><td>ABBREVIATED TITLE  - Qualifying information</td></tr><tr><td>245</td><td>h</td><td>TITLE STATEMENT  - Medium</td></tr><tr><td>246</td><td>5</td><td>VARYING FORM OF TITLE  - Institution to which field applies</td></tr><tr><td>340</td><td>3</td><td>PHYSICAL MEDIUM  - Materials specified</td></tr><tr><td>344</td><td>3</td><td>SOUND CHARACTERISTICS  - Materials specified</td></tr><tr><td>345</td><td>3</td><td>PROJECTION CHARACTERISTICS OF MOVING IMAGE  - Materials specified</td></tr><tr><td>346</td><td>3</td><td>VIDEO CHARACTERISTICS  - Materials specified</td></tr><tr><td>347</td><td>3</td><td>DIGITAL FILE CHARACTERISTICS  - Materials specified</td></tr><tr><td>382</td><td>e</td><td>MEDIUM OF PERFORMANCE  - Number of ensembles of the same type</td></tr><tr><td>382</td><td>n</td><td>MEDIUM OF PERFORMANCE  - Number of performers of the same medium</td></tr><tr><td>382</td><td>r</td><td>MEDIUM OF PERFORMANCE  - Total number of individuals performing alongside ensembles</td></tr><tr><td>382</td><td>s</td><td>MEDIUM OF PERFORMANCE  - Total number of performers</td></tr><tr><td>382</td><td>t</td><td>MEDIUM OF PERFORMANCE  - Total number of ensembles</td></tr><tr><td>382</td><td>v</td><td>MEDIUM OF PERFORMANCE  - Note</td></tr><tr><td>382</td><td>3</td><td>MEDIUM OF PERFORMANCE  - Materials specified</td></tr><tr><td>383</td><td>a</td><td>NUMERIC DESIGNATION OF MUSICAL WORK  - Serial number</td></tr><tr><td>383</td><td>b</td><td>NUMERIC DESIGNATION OF MUSICAL WORK  - Opus number</td></tr><tr><td>383</td><td>c</td><td>NUMERIC DESIGNATION OF MUSICAL WORK  - Thematic index number</td></tr><tr><td>383</td><td>e</td><td>NUMERIC DESIGNATION OF MUSICAL WORK  - Publisher associated with opus number</td></tr><tr><td>383</td><td>3</td><td>NUMERIC DESIGNATION OF MUSICAL WORK  - Materials secified</td></tr><tr><td>384</td><td>3</td><td>KEY  - Materials specified</td></tr><tr><td>500</td><td>5</td><td>GENERAL NOTE  - Institution to which field applies</td></tr><tr><td>501</td><td>5</td><td>WITH NOTE  - Institution to which field applies</td></tr><tr><td>502</td><td>o</td><td>DISSERTATION NOTE  - Dissertation identifier</td></tr><tr><td>505</td><td>u</td><td>FORMATTED CONTENTS NOTE  - Uniform Resource Identifier</td></tr><tr><td>506</td><td>5</td><td>RESTRICTIONS ON ACCESS NOTE  - Institution to which field applies</td></tr><tr><td>520</td><td>u</td><td>SUMMARY, ETC.  - Uniform Resource Identifier</td></tr><

#<CZTop::Socket::PUB:0x1ac82b0 last_endpoint="tcp://127.0.0.1:59847">